keyboards/sp111/rules.mk

```
@@ -0,0 +1,30 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes            # Smaller (and slightly faster) firmware
+
+
+# custom matrix setup
+CUSTOM_MATRIX = lite
+
+SRC += mcp23018.c matrix.c
+QUANTUM_LIB_SRC += i2c_master.c
```

Given what it does, I think I'd rather see the mcp code in the quantum lib part

```suggestion
SRC += matrix.c
QUANTUM_LIB_SRC += i2c_master.c mcp23018.c
```

---

Given the content of `mcp23018.c`, it should be fine to run lto against it (Given `i2c_master` is still linked as a separate unit).